src/order/conditionally_complete_lattice.lean

```diff
@@ -769,3 +769,182 @@ instance (α : Type*) [conditionally_complete_linear_order α] :
   ..order_dual.decidable_linear_order α }
 
 end order_dual
+
+section with_top_bot
+
+open lattice
+
+open_locale classical
+/-- Extends `Sup` from a conditionally complete lattice `α` to `with_top α`.
+The new `Sup` returns a non-junk value for more subsets (e.g. for unbounded-above subsets of `α`). -/
+noncomputable instance with_top.conditionally_complete_lattice.has_Sup
+  {α : Type*} [conditionally_complete_lattice α] : has_Sup (with_top α) :=
+⟨λ S, if ⊤ ∈ S then ⊤ else
+      let So := (coe ⁻¹' S : set α) in
+      if bdd_above So then ↑(Sup So) else ⊤⟩
+
+/-- Extends `Inf` from a conditionally complete lattice `α` to `with_top α`. -/
+noncomputable instance with_top.conditionally_complete_lattice.has_Inf
+  {α : Type*} [conditionally_complete_lattice α] : has_Inf (with_top α) :=
+⟨λ S, if S ⊆ {⊤} then ⊤ else ↑(Inf (coe ⁻¹' S) : α)⟩
+
+/-- Adding a top element to a conditionally complete lattice gives a conditionally complete lattice -/
+noncomputable instance with_top.conditionally_complete_lattice
+  {α : Type*} [conditionally_complete_lattice α] :
+  conditionally_complete_lattice (with_top α) :=
+{ le_cSup := λ S a hS haS,
+    let So := (coe ⁻¹' S : set α) in
+    show a ≤ (ite (⊤ ∈ S) ⊤ (ite (bdd_above So) ↑(Sup So) ⊤)),
```

Comment:
the readable version `if ... then ... else ...` is much easier to parse than the `ite` version, here and below.

Reply:
I've changed the ones which obviously make readability better. I have left the ones of the form `show a ≤ ite _ _ _,` but of course can change them if you think `if _ then _ else _` is more comprehensible.